executors/node/numberformat.js

```diff
@@ -80,36 +145,75 @@ module.exports = {
         options = none;
       }
     } else {
-      // Check each option for implementation.
+      // Set default maximumFractionDigits with these exceptions
+      if (options['notation'] && options['notation'] == 'compact' ||
+          options['style'] && options['style'] == 'currency' ||
+          options['maximumFractionDigits']) {
+        // Don't set maximumFractionDigits
+      } else {
+        options['maximumFractionDigits'] = 6;  // Default
+      }
```

Comment:
Adding `maximumFractionDigits: 6` should be in test generation, not in the executor